firewood/src/merkle/node.rs

```diff
@@ -214,9 +214,9 @@ impl From<NodeType> for Node {
 
 bitflags! {
     struct NodeAttributes: u8 {
-        const ROOT_HASH_VALID      = 0b001;
-        const IS_ENCODED_BIG_VALID = 0b010;
-        const LONG                 = 0b100;
+        const HAS_ROOT_HASH           = 0b001;
+        const ENCODED_LENGTH_IS_KNOWN = 0b010;
+        const ENCODED_IS_LONG         = 0b100;
```

Comment:
Why can't we remove `ENCODED_IS_LONG` as if not `ENCODED_LENGTH_IS_KNOWN` then it is longer than hash value.

Reply:
`ENCODED_LENGTH_IS_KNOWN` isn't the best name, but right now, there's a case where `Node` has a value for `is_encoded_longer_than_hash` but doesn't have a value for `encoded` or `root_hash`. I'm not sure if there's code that would lead to this being the case, but that context shouldn't be captured in here. So we definitely need three separate states, or `Node` itself should be changed.

Reply:
That should never happen as `is_encoded_longer_than_hash` is generated based on the comparison between `encoded` and `root_hash`. But I see your point. However, this makes this simplifying not so meaningful. The result code is a bit more confusing..

Reply:
>The result code is a bit more confusing

Before, `IS_ENCODED_BIG_VALID` is ONLY set when we've *already* calculated the length of `encoded` and know that the length is **smaller** than that of a hash. I don't see how you could possibly make things more confusing than that haha.

Now, we have one flag for whether or not we know the length of the hash and another flag for whether the hash is long. I don't see how that's more confusing. 

However, I think the way that we're both looking at this is wrong. We are coupling node-logic and serialization-logic. If we manually create an invalid node with `is_encoded_length_longer_than_hash`, when the `encoded` value is actually shorter, the node should still serialize and deserialize to the same thing. 

I think the flags should remain as they currently are, but I will change the logic.